lib/pool.py

```diff
@@ -17,6 +17,14 @@ def __init__(self, master_hostname_or_ip):
         assert master.is_master(), f"Host {master_hostname_or_ip} is not a master host. Aborting."
         self.master = master
         self.hosts = [master]
+
+        # wait for XAPI startup to be done, or we can get "Connection
+        # refused (calling connect )" when calling self.hosts_uuids()
+        wait_for(lambda: commands.ssh(master_hostname_or_ip, ['xapi-wait-init-complete', '60'],
+                                      check=False, simple_output=False).returncode == 0,
+                 f"Wait for XAPI init to be complete on {master_hostname_or_ip}",
+                 timeout_secs=30 * 60)
```

Comment:
This could be useful in other places, if this implementation is better than the existing one which does a similar check by waiting for `xe` to return that the current host is enabled, as we do each time we reboot a host or restart XAPI.

I don't know which is safer:
* I'm not 100% confident in this `xapi-wait-init-complete` check
* I've seen the enabled status displayed by XO flip to enable, then shortly to disabled again, then enabled again for good.

I'd start with making it a helper method of the Host class, and decide later in what direction to harmonize (maybe both checks added together?)

Reply:
> This could be useful in other places

In theory yes, but in practice, how often do we interact with XAPI without building a `Pool` object?

> I'm not 100% confident in this xapi-wait-init-complete check

Neither do I, but I have not seen any practical issue with it, and it does fix those flakiness issues AFAICT.

> I've seen the enabled status displayed by XO flip to enable, then shortly to disabled again, then enabled again for good.

That would look like a bug in XAPI